src/Polynomials.jl

```diff
@@ -348,14 +348,24 @@ end
 div(num::Poly, den::Poly) = divrem(num, den)[1]
 rem(num::Poly, den::Poly) = divrem(num, den)[2]
 
-function ==(p1::Poly, p2::Poly)
-    if p1.var != p2.var
-        return false
-    else
-        return p1.a == p2.a
-    end
+==(p1::Poly, p2::Poly) = (p1.var == p2.var && p1.a == p2.a)
+==(p1::Poly, n::Number) = (coeffs(p1) == [n])
+==(n::Number, p1::Poly) = (p1 == n)
+
+function isapprox{T,S}(p1::Poly{T}, p2::Poly{S};
+  rtol::Real = sqrt(eps(promote_type(T,S))), atol::Real = 0)
```

Comment:
Hmm, there is some thinking to do here. My first inclination would to truncate before calling isapprox, and then return `false` if they aren't the same. I don't really like throwing an error here, as that seems unexpected, especially as we allow the comparison of polys to numbers. Likely it is beset to merge this in first and discuss these issues separately.